client/state/sites/plans/assembler.js

```diff
@@ -9,6 +9,7 @@ const createSitePlanObject = ( plan ) => {
 	}
 
 	return {
+		auto_renew: Boolean( plan.auto_renew ),
```

Comment:
From D2621-code it sounds like there are some technical hurdles, so this value will be incorrect for plans paid with credits. Do you mind adding a comment here that explains this.